Core/include/Acts/Utilities/AngleHelpers.hpp

```diff
@@ -9,26 +9,120 @@
 #pragma once
 
 #include <cmath>
+#include <numbers>
 
 namespace Acts::AngleHelpers {
 
+template <typename Scalar>
+struct EtaThetaConversionTraits {};
+
+template <>
+struct EtaThetaConversionTraits<float> {
+  static constexpr float minTheta = 1e-6f;
+  static constexpr float maxTheta = std::numbers::pi_v<float> - 1e-6f;
+
+  static constexpr float minEta = -80.0f;
+  static constexpr float maxEta = 80.0f;
+};
+
+template <>
+struct EtaThetaConversionTraits<double> {
+  static constexpr double minTheta = 1e-12;
+  static constexpr double maxTheta = std::numbers::pi - 1e-12;
+
+  static constexpr double minEta = -700.0;
+  static constexpr double maxEta = 700.0;
+};
+
+template <typename Scalar>
+struct ClampedEtaThetaConversionTraits {
+  // computed from minEta
+  static constexpr Scalar minTheta =
+      static_cast<Scalar>(3.6097027756908306e-35);
+  // computed from maxEta
+  static constexpr Scalar maxTheta = static_cast<Scalar>(3.141592653589793);
+
+  static constexpr Scalar minEta = static_cast<Scalar>(-80.0);
+  static constexpr Scalar maxEta = static_cast<Scalar>(80.0);
```

Comment:
A `float` `maxTheta=3.141592653589793` gives `nan` when converted to &eta;. I think it would be better to clamp at &eta;=10.

```suggestion
  static constexpr Scalar minEta = static_cast<Scalar>(-10.0);
  static constexpr Scalar maxEta = static_cast<Scalar>(10.0);
```
and update `minTheta` and `maxTheta` above:
```
  static constexpr Scalar minTheta = static_cast<Scalar>(9.0799859462585550026e-05L);
  static constexpr Scalar maxTheta = static_cast<Scalar>(3.1415018537303306529L);
```

Reply:
I cannot reproduce the `nan`. Isn't the value you are referring to checked against in `etaFromThetaClamed` so it has to produce `minEta`?

Reply:
I checked with the non-clamped formula. The thing is, I don't know if the next-smallest value gives `nan`. I thought 10 would be safer, but if you are sure 80 is OK for `float`, then that's fine.

Reply:
opted for 10 now as the theta scale looks more appropriate for healthy floats